modules/core/src/main/scala/org/tessellation/domain/snapshot/SnapshotReference.scala

```diff
@@ -0,0 +1,15 @@
+package org.tessellation.domain.snapshot
+
+import cats.data.NonEmptySet
+
+import org.tessellation.kernel.StateChannelSnapshot
+import org.tessellation.schema.height.Height
+import org.tessellation.schema.peer.PeerId
+
+case class SnapshotReference(
```

Comment:
So after today's discussion, I think that `SnapshotReference` should contain only the `Height` as it's triggered by tips. The other trigger should produce some `case object` just to signal the occurrence of the trigger (even the `Unit` type would work, but for clarity, I would go with `case object`).